baselines/fedNova/README.md

```diff
@@ -0,0 +1,87 @@
+
+
+## About this baseline
+
+****What’s implemented:**** *The code in this baseline aims to reproduce the results from Fig 5 in the paper which corresponds to Synthetic(1,1) experiment and further Table 1 for CIFAR 10 dataset.*
+
+****Datasets:**** *_This basleline experiments tackles heterogeneous data sources. Non-IID CIFAR-10 dataset and Synthetic(1,1) dataset(creation methodology detailed in [link](https://arxiv.org/pdf/1812.06127.pdf) are used in the experiments._*
+
+****Hardware Setup:**** :warning: *_Give some details about the hardware (e.g. a server with 8x V100 32GB and 256GB of RAM) you used to run the experiments for this baseline. Someone out there might not have access to the same resources you have so, could list the absolute minimum hardware needed to run the experiment in a reasonable amount of time ? (e.g. minimum is 1x 16GB GPU otherwise a client model can’t be trained with a sufficiently large batch size). Could you test this works too?_*
+
+****Contributors:**** :warning: *_Aasheesh Singh(Github: @ashdtu), MILA-Quebec AI Institute_*
+
+
+## Experimental Setup
+
+****Task:**** :warning: *_what’s the primary task that is being federated? (e.g. image classification, next-word prediction). If you have experiments for several, please list them_*
+
+****Model:**** :warning: *_provide details about the model you used in your experiments (if more than use a list). If your model is small, describing it as a table would be :100:. Some FL methods do not use an off-the-shelve model (e.g. ResNet18) instead they create your own. If this is your case, please provide a summary here and give pointers to where in the paper (e.g. Appendix B.4) is detailed._*
+
+****Dataset:**** :warning: *_Earlier you listed already the datasets that your baseline uses. Now you should include a breakdown of the details about each of them. Please include information about: how the dataset is partitioned (e.g. LDA with alpha 0.1 as default and all clients have the same number of training examples; or each client gets assigned a different number of samples following a power-law distribution with each client only instances of 2 classes)? if  your dataset is naturally partitioned just state “naturally partitioned”; how many partitions there are (i.e. how many clients)? Please include this an all information relevant about the dataset and its partitioning into a table._*
+
+****Training Hyperparameters:**** :warning: *_Include a table with all the main hyperparameters in your baseline. Please show them with their default value._*
+
+
+## Environment Setup
+
```

Comment:
I know you are still working on this but let me add some content that would be great to have (this is what essentially other baselines are doing) -- add it as a code block:
```
# Set python version
pyenv install 3.10.6
pyenv local 3.10.6

# Tell poetry to use python 3.10
poetry env use 3.10.6

# install the base Poetry environment
poetry install

# activate the environment
poetry shell
```